homeassistant/components/aemet/sensor.py

```diff
@@ -232,108 +354,61 @@ async def async_setup_entry(
+    """Implementation of an AEMET OpenData sensor."""
 
     _attr_attribution = ATTRIBUTION
+    entity_description: AemetSensorEntityDescription
 
     def __init__(
         self,
-        name,
-        unique_id,
+        name: str,
         coordinator: WeatherUpdateCoordinator,
-        description: SensorEntityDescription,
+        description: AemetSensorEntityDescription,
+        config_entry: ConfigEntry,
     ) -> None:
         """Initialize the sensor."""
         super().__init__(coordinator)
         self.entity_description = description
         self._attr_name = f"{name} {description.name}"
-        self._attr_unique_id = unique_id
-
-
-class AemetSensor(AbstractAemetSensor):
-    """Implementation of an AEMET OpenData sensor."""
-
-    def __init__(
-        self,
-        name,
-        unique_id_prefix,
-        weather_coordinator: WeatherUpdateCoordinator,
-        description: SensorEntityDescription,
-    ) -> None:
-        """Initialize the sensor."""
-        super().__init__(
-            name=name,
-            unique_id=f"{unique_id_prefix}-{description.key}",
-            coordinator=weather_coordinator,
-            description=description,
-        )
-
-    @property
-    def native_value(self):
-        """Return the state of the device."""
-        return self.coordinator.data.get(self.entity_description.key)
-
-
-class AemetForecastSensor(AbstractAemetSensor):
-    """Implementation of an AEMET OpenData forecast sensor."""
-
-    def __init__(
-        self,
-        name,
-        unique_id_prefix,
-        weather_coordinator: WeatherUpdateCoordinator,
-        forecast_mode,
-        description: SensorEntityDescription,
-    ) -> None:
-        """Initialize the sensor."""
-        super().__init__(
-            name=name,
-            unique_id=f"{unique_id_prefix}-{description.key}",
-            coordinator=weather_coordinator,
-            description=description,
-        )
-        self._forecast_mode = forecast_mode
-        self._attr_entity_registry_enabled_default = (
-            self._forecast_mode == FORECAST_MODE_DAILY
-        )
+        self._attr_unique_id = f"{config_entry.unique_id}-{description.key}"
 
     @property
     def native_value(self):
         """Return the state of the device."""
-        forecast = None
-        forecasts = self.coordinator.data.get(
-            FORECAST_MODE_ATTR_API[self._forecast_mode]
-        )
-        if forecasts:
-            forecast = forecasts[0].get(self.entity_description.key)
-            if self.entity_description.key == ATTR_API_FORECAST_TIME:
-                forecast = dt_util.parse_datetime(forecast)
-        return forecast
+        value = self.get_aemet_value(self.entity_description.keys)
+        if self.entity_description.keys[-1] == AOD_CONDITION:
+            value = CONDITIONS_MAP.get(value)
+        elif self.entity_description.keys[-1] == AOD_TIMESTAMP:
+            value = dt_util.parse_datetime(value)
+        return value
```

Comment:
```suggestion
        return self.entity_description.value_fn(value)
```

Reply:
Done in f064f9e010966a5c9832823cc324e04c7e348d2a